services/core/PlatformDriverAgent/HomeAssistantDriver.md

```diff
@@ -0,0 +1,239 @@
+```
+
+**Controlling Entities**
+
+Controlling Home Assistant entities happens in the parse_config method.
+
+As before, we need to create headers and a URL for use. This time we use light/turn_on at the end of the URL, the "light" is the domain and "turn_on" is the service. You can read more here https://developers.home-assistant.io/docs/api/rest/ 
+
+Next, we add the entity_id and brightness in the payload to add it to our response, we also use requests.post rather than requests.get. 
+When we call the function, it will turn on all the lights and set the brightness. Control for the thermostats is pretty much the same. 
+
+We also assume that light entities start with "light" since in Home Assistant all light entites will indeed start with the string light. This allows us to only send controls for lights rather than sending the turn_on service to other devices. The same has been done for thermostats. 
+
+```python
+        def turn_on_lights(brightness_level):
+            url2 = f"http://{self.ip_address}:{self.port}/api/services/light/turn_on"
+            
+            headers = {
+                    "Authorization": f"Bearer {self.access_token}",
+                    "Content-Type": "application/json",
+                }
+            point_names = point_name.split('\n')
+            for entity in point_names:
+                if entity.startswith("light"): # this will ensure that only lights are conrolled and not other devices
+                    try:
+                     # ranges from 0 - 255 for most lights
+                        payload = {
+                            "entity_id": f"{entity}",
+                            "brightness": brightness_level,
+                        }
+                        response = requests.post(url2, headers=headers, data=json.dumps(payload))
+                        if response.status_code == 200:
+                            _log.info(f"Turned on {entity}")
+                    except:
+                        continue
+                else:
+                    continue
+```
+
+**Reaacting to changes**
+
+Reacting to changes happens in the _create_subscriptions and _handle_publish functions. 
+
+```python
+    def _create_subscriptions(self, topic):
+        """
+        Unsubscribe from all pub/sub topics and create a subscription to a topic in the configuration which triggers
+        the _handle_publish callback
+        """
+        self.vip.pubsub.unsubscribe("pubsub", None, None)
+
+        self.vip.pubsub.subscribe(peer='pubsub',
+                                  prefix=topic,
+                                  callback=self._handle_publish)    
+        
+    def _handle_publish(self, peer, sender, bus, topic, headers, messages):
+        for message in messages: #subscribes to itself then stores states and if they change it runs the commands 
+            for entity_id, entity_data in message.items():
+                
+                state = entity_data.get("state", None)
```

Comment:
You have a lot of assumptions about an entity_id.  I think that it would be good to document those outside of just the code here.